### PR TITLE
Fix sycl shared build

### PR DESCRIPTION
**Pull request by waynemitchell**
This fixes the shared library build for hypre with the sycl backend.

### CODE REVIEW

File: src/configure [hunk 1]
```diff
@@ -4582,30 +4577,6 @@ then :
...
-
-
-
-# Check whether --with-magma-lib was given.
-if test "${with_magma_lib+set}" = set; then :
-  withval=$with_magma_lib; for magma_lib in $withval; do
-    MAGMA_LIBS="$MAGMA_LIBS $magma_lib"
- done
-
-fi
```
Comment (waynemitchell) [inline src/configure:4608]: @victorapm , when I reran `config/update.sh`, there were some deletions of magma-related stuff. This stuff mostly looks redundant, but I wanted to double check with you that this looks OK.

Comment (waynemitchell) [reply src/configure:4608]: Ah, these deletions actually break the build. I get the error:

icpx: error #10005: unable to open argument file 'MAGMA_LIBS@'

I notice that we have both `MAGMA_LIBS` and `HYPRE_MAGMA_LIB` listed in Makefile.config.in:

https://github.com/hypre-space/hypre/blob/master/src/config/Makefile.config.in#L177

Should one of these sets of magma-related variables be removed?

Comment (victorapm) [reply src/configure:4608]: Hi Wayne, yes, the ones without the hypre prefix should be removed.

I thought i had fixed this before... Sorry..

**Review by victorapm — APPROVED** [review]

**Review by ulrikeyang — APPROVED** [review]
This looks good, as far as I can tell. Thanks for taking care of the MAGMA inclusions also. We want those to work.

File: src/configure (hunk 1 above)
Comment (waynemitchell) [reply src/configure:4608]: No problem. You can double check my final changes. I removed `MAGMA_LIBS` replacing with `HYPRE_MAGMA_LIB` and `HYPRE_MAGMA_LIB_DIR` where appropriate.

Comment (osborn9) [reply src/configure:4608]: I just tested a MAGMA enabled build and everything works as expected.

**Review by rfalgout — APPROVED** [review]
Looks good to me.  Thanks!

**Review by osborn9 — APPROVED** [review]
Looks good! 

Note - there are still some issues with the spack build, but are unrelated to this change (and maybe even hypre proper)

**Pull request merged**